ScanDoc test app

avargic opened the pull request:
AGP & gradle version bump
Add cameraX dependency
Add scandoc functionality to test app with 3 new fragments:

1. camera preview fragment - for capturing image of card
2. image preview fragment - for checking the captured image before sending to server for validation and extraction
3. extracted data fragment - for displaying the data extracted from the image, as well as displaying the image that was received and processed by server

avargic committed:
Finishing scandoc implementation
Add scandoc functionality to test app
AGP & gradle version bump
Add cameraX dependency

avargic committed:
Add unit tests
Add kotlin configuration
Add assertJ dependency
Minor fixes
Add missing getters
Add missing response json properties

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/ScanDocCameraPreviewFragment.java:
```diff
@@ -0,0 +1,145 @@
+
+    private void captureImage() {
+        final File photoFile = new File(getActivity().getExternalMediaDirs()[0], String.format(FILE_NAME_FORMAT, System.currentTimeMillis()));
+        ImageCapture.OutputFileOptions outputOptions = new ImageCapture.OutputFileOptions.Builder(photoFile).build();
```
final

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/ScanDocCameraPreviewFragment.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    private void captureImage() {
+        final File photoFile = new File(getActivity().getExternalMediaDirs()[0], String.format(FILE_NAME_FORMAT, System.currentTimeMillis()));
```
Ovaj 0 staviti u konstantu da se zna cemu sluzi

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/ScanDocImagePreviewFragment.java:
```diff
@@ -0,0 +1,152 @@
+            false,
+            false,
+            false,
+            false
```
Ovo isto staviti u konstante tako da se zna sto je koji parametar.

luka-dragutin-asee commented on app/src/main/java/com/monri/android/example/ScanDocImagePreviewFragment.java:
```diff
@@ -0,0 +1,152 @@
+
+    private void validateImage() {
+        progressBar.setVisibility(View.VISIBLE);
+        scanDocApi.validateScannedCard(base64Image, false, new ScanDocValidationCallback(), List.of());
```
`Collections.emptyList()`

avargic committed:
Rename .java to .kt

avargic committed:
Add missing final modifiers
Refactoring httpsClient to return HttpsRequest
Add httpsClientProxy
Moving Monri urls to url provider
Configure kotlin for application module

avargic committed:
Changes to align with ios sdk

avargic committed:
Code review fixes + add validation/extraction methods that take image as byte array

avargic committed:
Code review fixes

avargic committed:
Code review fixes

avargic committed:
Monri SDK version bump
Update readme

avargic committed:
Merge pull request #49 from MonriPayments/feature/scandocTests
